<launch>
	<arg name="camera"  default="camera" />
	
	<arg name="sigverse_ros_bridge_port"        default="50001" />
	<arg name="ros_bridge_port"                 default="9090" />

	<node name="grasping_auto" pkg="sigverse_turtlebot3_open_manipulator" type="grasping_auto" output="screen" launch-prefix="gnome-terminal -t 'TurtleBot3 teleop key' -- ">
		<!--
		<remap from="/tb3omc/joint_state"         to="/tb3omc/joint_state" />
		<remap from="/tb3omc/cmd_vel"             to="/tb3omc/cmd_vel" />
		<remap from="/tb3omc/joint_trajectory"    to="/tb3omc/joint_trajectory" />
		<remap from="/camera/rgb/camera_info"     to="/camera/rgb/camera_info" />
		<remap from="/camera/depth/points"        to="/camera/depth/points" />
		<remap from="/darknet_ros/bounding_boxes" to="/darknet_ros/bounding_boxes" />
		-->
	</node>

	<!-- Config and weights folder. -->
	<arg name="yolo_weights_path"          default="$(find darknet_ros)/yolo_network_config/weights"/>
	<arg name="yolo_config_path"           default="$(find darknet_ros)/yolo_network_config/cfg"/>

	<!-- Load parameters -->
	<rosparam command="load" ns="darknet_ros" file="$(find darknet_ros)/config/ros.yaml"/>
	<rosparam command="load" ns="darknet_ros" file="$(find darknet_ros)/config/yolov2-tiny.yaml"/>

	<!-- Start darknet and ros wrapper -->
	<node pkg="darknet_ros" type="darknet_ros" name="darknet_ros" output="screen">
		<param name="weights_path"          value="$(arg yolo_weights_path)" />
		<param name="config_path"           value="$(arg yolo_config_path)" />
		
		<param name="subscribers/camera_reading/topic"  value="/$(arg camera)/rgb/image_raw" />
	</node>

	<group ns="sigverse_ros_bridge">
		<node name="sigverse_ros_bridge" pkg="sigverse_ros_bridge" type="sigverse_ros_bridge" args="$(arg sigverse_ros_bridge_port)"/>
	</group>
	
	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" > 
		<arg name="port" value="$(arg ros_bridge_port)"/>
	</include>

	<node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" />

	<node pkg="nodelet" type="nodelet" name="nodelet1" args="load depth_image_proc/point_cloud_xyz nodelet_manager">
		<remap from="camera_info" to="/$(arg camera)/depth/camera_info" />
		<remap from="image_rect"  to="/$(arg camera)/depth/image_raw" />
		<remap from="points"      to="/$(arg camera)/depth/points" />
	</node>
  
	<node pkg="image_view" type="image_view" name="image_view" args="image:=$(arg camera)/rgb/image_raw"/>
</launch>

